### PR TITLE
LoadFITS Improved Memory usage and removed LoadAsRectangle property

### LINKED ISSUE #18242: Imaging Framework: Make LoadFits more memory and time efficient
Find ways to make LoadFits more space efficient
- Currenly loading 100 of 8MB images takes up ~10GB of RAM
  - We fill up a X and E dimension (removed example:  change in readDataToImgs, allows us to save 2/3 of the space used, however we lose the X data which is the position of the row in real cms, however that could be easily calculated as it is the row number * pixel size : https://gist.github.com/DimitarTasev/d236ec80563ba8a6a39d7615a9bfc67f)
   - We use doubles, instead of int16, which means 3/4 of data loss, or float32, which is uses only twice as much RAM
   - Having the X and E removed means we reduce our data by 2/3, having the int16 means we further reduce it by 3/4, coming down to 10 * 0.33 * 0.25 = ~900MB of RAM, instead of 10GB

### PR DESCRIPTION
**Pull request by dtasev**
Description of work.
This makes `LoadFITS` use the `sharedX` and `sharedE` HistogramData functionality to greatly reduce the memory consumption.

Additionally this PR hides the property for `LoadAsRectangle` because a single 16MB image was loaded into a 7GB workspace. No fix was added for that as `LoadFITS` might be deprecated and removed soon.
16mb_image_2minutes_7gigs
 

**To test:**
- Run LoadFITS test
- Try loading a Fits image, here are a few: https://drive.google.com/open?id=0B2Mj8e56Z8xTNTZRV1NuSWs5QVU

Fixes #18242.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/DataHandling/src/LoadFITS.cpp [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
 #include "MantidAPI/NumericAxis.h"
 #include "MantidAPI/RegisterFileLoader.h"
 #include "MantidAPI/Run.h"
 #include "MantidAPI/WorkspaceFactory.h"
 #include "MantidAPI/WorkspaceGroup.h"
-#include "MantidDataHandling/LoadFITS.h"
 #include "MantidDataObjects/Workspace2D.h"
+#include "MantidHistogramData/Points.h"
+#include "MantidHistogramData/LinearGenerator.h"
 #include "MantidKernel/BoundedValidator.h"
```
Comment (AnthonyLim23) [inline Framework/DataHandling/src/LoadFITS.cpp]: LinearGenerator should be beforePoints

File: Framework/DataHandling/src/LoadFITS.cpp [hunk 2]
```diff
@@ -121,12 +123,12 @@ void LoadFITS::init() {
...
-                                                   Kernel::Direction::Input),
-      "If enabled (not by default), the output Workspace2D will have "
-      "one histogram per row and one bin per pixel, such that a 2D "
-      "color plot (color fill plot) will display an image.");
+  // declareProperty(
+  //     make_unique<Kernel::PropertyWithValue<bool>>("LoadAsRectImg", true,
+  //                                                  Kernel::Direction::Input),
+  //     "If enabled (not by default), the output Workspace2D will have "
+  //     "one histogram per row and one bin per pixel, such that a 2D "
+  //     "color plot (color fill plot) will display an image.");
```
Comment (AnthonyLim23) [inline Framework/DataHandling/src/LoadFITS.cpp]: why not just delete this?

**Comment by AnthonyLim23** [comment]
I have used the loadFITS algorithm and had no problems loading the data provided. However, the performance test fails (unhandled exception)

**Comment by DTasev** [comment]
With the GUI development, this algorithms will be soon removed as well as the current Tomo GUI. I am closing this PR for that reason

**Pull request closed without merging**